test/token/ERC1155/ERC1155Base.behavior.js

```diff
@@ -108,11 +116,39 @@ const describeBehaviorOfERC1155Base = function ({ deploy }, skips) {
     });
 
     describe('#safeTransferFrom', function () {
-      it('todo');
+      it('sends amount from A to B', async function(){
+        const id = ethers.constants.Zero;
+        const amount = ethers.constants.Two;
+
+        expect(await instance.callStatic.balanceOf(holder.address, id)).to.equal(0);
+
+        await mint(holder.address, id, amount);
+
+        expect(await instance.callStatic.balanceOf(holder.address, id)).to.equal(amount);
+
+        await instance.connect(holder).safeTransferFrom(holder.address, spender.address, id, amount, ethers.utils.randomBytes(0));
+
+        expect(await instance.callStatic.balanceOf(holder.address, id)).to.equal(0);
+        expect(await instance.callStatic.balanceOf(spender.address, id)).to.equal(amount);
+      });
     });
 
     describe('#safeBatchTransferFrom', function () {
-      it('todo');
+      it('sends amount from A to B, batch version', async function(){
+        const id = ethers.constants.Zero;
+        const amount = ethers.constants.Two;
+
+        expect(await instance.callStatic.balanceOfBatch([holder.address], [id])).to.deep.have.members([ethers.constants.Zero]);
+
+        await mint(holder.address, id, amount);
+
+        expect(await instance.callStatic.balanceOfBatch([holder.address], [id])).to.deep.have.members([amount]);
+
+        await instance.connect(holder).safeBatchTransferFrom(holder.address, spender.address, [id], [amount], ethers.utils.randomBytes(0));
```

Comment:
Use the `changeTokenBalances` matcher here: https://ethereum-waffle.readthedocs.io/en/latest/matchers.html

No need to check that initial balance is 0.

Reply:
Disregard this.  The matcher only works with ERC20 balances.